modules/parser/src/parsers/parse-xviz-message-sync.js

```diff
@@ -37,15 +39,17 @@ export function parseXVIZMessageSync(message, onResult, onError, opts) {
     return;
   }
 
+  const {messageType} = opts;
+
   try {
-    const xvizData = new XVIZData(message);
+    const xvizData = new XVIZData(message, {messageType});
     const xvizMsg = xvizData.message();
 
     // Non-xviz messages will return null
     if (xvizMsg) {
       const data = xvizMsg.data;
 
-      const v2Type = xvizMsg.type || undefined;
+      const v2Type = messageType || xvizMsg.type || undefined;
```

Comment:
If we are passing XVIZData messageType, wouldnt the result from `xvizData.message()` reflect that, such that we don't need to change this line